fcrepo-http-api/src/main/java/org/fcrepo/http/api/responses/StreamingBaseHtmlProvider.java

```diff
@@ -187,35 +186,26 @@ void init() throws IOException {
     }
 
     @Override
-    public void writeTo(final RdfStream rdfStream, final Class<?> type,
+    public void writeTo(final RdfNamespacedStream nsStream, final Class<?> type,
                         final Type genericType, final Annotation[] annotations,
                         final MediaType mediaType,
                         final MultivaluedMap<String, Object> httpHeaders,
                         final OutputStream entityStream) throws IOException {
 
-        try {
-            final RdfStream nsRdfStream = new NamespaceRdfContext(rdfStream.session());
-
-            rdfStream.namespaces(nsRdfStream.namespaces());
-
-            final Node subject = VIEW_HELPERS.getContentNode(rdfStream.topic());
-
-            final Model model = rdfStream.asModel();
+        final Node subject = VIEW_HELPERS.getContentNode(nsStream.stream.topic());
```

Comment:
Implementing `RdfStream` would require considerably more code. You would need to have `RdfNamespacedStream extend RdfStream` and then have an implementing class along the lines of `DefaultRdfStream`. I would really prefer to stick with my method -- it's just a Tuple.